### PR TITLE
ETFE-2472: Add trader name/ERN and 'change trader link' to every page…

### PR DESCRIPTION
**Pull request by nickg-hmrc**
…, if user has multiple ERNs on their login

### CODE REVIEW

File: .g8/multipleQuestionsPage/app/views/$className$View.scala.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 @(form: Form[_], mode: Mode)(implicit request: DataRequest[_], messages: Messages)
 
-@layout(pageTitle = title(form, messages("$className;format="decap"$.title"))) {
+@layout(pageTitle = title(form, messages("$className;format="decap"$.title")), , maybeShowActiveTrader = maybeShowActiveTrader(request)) {
```
Comment (LewisMitchell) [inline .g8/multipleQuestionsPage/app/views/$className$View.scala.html]:
```suggestion
@layout(pageTitle = title(form, messages("$className;format="decap"$.title")), maybeShowActiveTrader = maybeShowActiveTrader(request)) {
```

File: conf/application.conf
Comment (LewisMitchell) [inline conf/application.conf:1]: There seems to be a missing config entry for `emcs-tfe-reference-data`

File: test/views/components/ActiveTraderSpec.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        "link must exist" in {
+          val html = activeTrader(Some(traderInfo), appConfig)
+          val doc = Jsoup.parse(html.toString())
+
+          val link =  doc.select(linkSelector).first
+
+          (link.attr("href"), link.text) mustEqual ("http://localhost:8310/emcs-tfe", messagesForLanguage.changeTraderType)
+        }
+
+      }
```
Comment (LewisMitchell) [inline test/views/components/ActiveTraderSpec.scala]: Same as the other PR, could squash this into one test

File: test/connectors/referenceData/GetTraderKnownFactsHttpParserSpec.scala
Comment (LewisMitchell) [inline test/connectors/referenceData/GetTraderKnownFactsHttpParserSpec.scala:1]: Needs a test for bad JSON

File: conf/application.conf [hunk 3]
```diff
@@ -45,26 +45,31 @@ play.http.errorHandler = "handlers.ErrorHandler"
...
+    user-allow-list {
+      protocol = http
+      host = localhost
+      port = 11306
+    }
+    emcs-tfe-reference-data-stub {
+      protocol = http
+      host = localhost
+      port = 8309
+    }
```
Comment (LewisMitchell) [inline conf/application.conf]: QA/Staging/Production config needed please

File: conf/application.conf [hunk 4]
```diff
@@ -87,6 +92,7 @@ timeout-dialog {
 features {
   welsh-translation = false
   allowListEnabled = false
+  stub-get-trader-known-facts = true
```
Comment (LewisMitchell) [inline conf/application.conf:100]: QA/Staging/Production config needed please

File: app/connectors/referenceData/GetTraderKnownFactsConnector.scala
Comment (LewisMitchell) [inline app/connectors/referenceData/GetTraderKnownFactsConnector.scala:1]: Could you add an integration test for this, please?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: .g8/multipleQuestionsPage/app/views/$className$View.scala.html (hunk 1 above)
Comment (nickg-hmrc) [reply .g8/multipleQuestionsPage/app/views/$className$View.scala.html]: good catch

File: conf/application.conf (hunk 3 above)
Comment (nickg-hmrc) [reply conf/application.conf]: Here are the app-config-* changes:
https://github.com/hmrc/app-config-base/pull/9709
https://github.com/hmrc/app-config-production/pull/20426
https://github.com/hmrc/app-config-qa/pull/20664
https://github.com/hmrc/app-config-staging/pull/14220

File: app/config/AppConfig.scala [hunk 5]
```diff
@@ -109,6 +109,14 @@ class AppConfig @Inject()(servicesConfig: ServicesConfig, configuration: Configu
...
 
+  private def traderKnownFactsReferenceDataService: String =
+    if (isEnabled(StubGetTraderKnownFacts)) {
+      servicesConfig.baseUrl("emcs-tfe-reference-data-stub")
+    }
+    else {
+      servicesConfig.baseUrl("emcs-tfe-reference-data")
+    }
 
+  def traderKnownFactsReferenceDataBaseUrl: String = s"$traderKnownFactsReferenceDataService/emcs-tfe-reference-data"
```
Comment (LewisMitchell) [inline app/config/AppConfig.scala:121]: Could you add a test in AppConfigSpec? Similar to how you did it [here](https://github.com/hmrc/emcs-tfe-cancel-movement-frontend/blob/bba0881290635eb3ee5d54976089bc20727e9de1/test/config/AppConfigSpec.scala#L108-L126)

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**